app.py

```diff
@@ -1,73 +1,82 @@
 from flask import Flask, render_template, request, redirect, url_for
 import trello_service as trello
 import logging
-
-app = Flask(__name__)
-app.config.from_object('flask_config.Config')
-
-
-@app.route('/favicon.ico')
-def favicon():
-    return app.send_static_file('web/favicon.ico')
-
-
-@app.route('/')
-def index():
-    cards = get_and_sort_cards_by_status()
-    return render_template('index.html', items=cards)
-
-
-@app.route('/', methods=['POST'])
-def add_item():
-
-    title = request.form['title']
-    cards = trello.get_cards()
-
-    if title == '':
+from view_model import ViewModel
+from card import Status
+
+
+def create_app():
+    app = Flask(__name__)
+    app.config.from_object('flask_config.Config')
+
+    @app.route('/favicon.ico')
+    def favicon():
+        return app.send_static_file('web/favicon.ico')
+
+    @app.route('/')
+    def index():
+        cards = get_and_sort_cards_by_status()
+        view_model = ViewModel(cards)
+        return render_template('index.html', view_model=view_model,
+                               status_done=Status.DONE,
+                               status_todo=Status.TODO,
+                               status_doing=Status.DOING)
+
+    @app.route('/', methods=['POST'])
+    def add_item():
+
+        title = request.form['title']
+        cards = get_and_sort_cards_by_status()
+
+        if title == '':
+            view_model = ViewModel(cards)
+            view_model.error = "Title can not be empty"
+            return render_template('index.html', view_model=view_model,
+                                   status_done=Status.DONE,
+                                   status_todo=Status.TODO,
+                                   status_doing=Status.DOING)
```

Comment:
You could actually still use redirect('/') (to avoid duplicating code & keep browser behaviour correct). You instead use Flask's `flash` to display messages in the way you want.